cluster.go

```diff
@@ -550,57 +636,57 @@ func (c *Cluster) fragCombos(idx string, maxSlice uint64, frameViews viewsByFram
 // diff compares c with another cluster and determines if a node is being
 // added or removed. An error is returned for any case other than where
 // exactly one node is added or removed.
-func (c *Cluster) diff(other *Cluster) (action string, uri URI, err error) {
+func (c *Cluster) diff(other *Cluster) (action string, nodeID string, err error) {
 	lenFrom := len(c.Nodes)
 	lenTo := len(other.Nodes)
 	// Determine if a node is being added or removed.
 	if lenFrom == lenTo {
-		return action, uri, errors.New("clusters are the same size")
+		return action, nodeID, errors.New("clusters are the same size")
```

Comment:
this applies elsewhere as well, but I think in general it's best to explicitly return 0 values for everything when you're returning a non-nil error. i.e. `return "", "", errors.New(...)`